contracts/Allocator.sol

```diff
@@ -420,43 +449,68 @@ contract Allocator is Permissions, IERC777Recipient {
-        uint256 date = now;
+        
         Beneficiary memory beneficiaryPlan = _beneficiaries[wallet];
         Plan memory planParams = _plans[beneficiaryPlan.planId - 1];
-        if (date < timeHelpers.addMonths(beneficiaryPlan.startMonth, planParams.vestingCliff)) {
+
+        uint256 firstVestingMonth = beneficiaryPlan.startMonth.add(planParams.vestingCliff);
+        if (now < timeHelpers.monthToTimestamp(firstVestingMonth)) {
             return 0;
+        } else {
+            uint256 currentMonth = timeHelpers.getCurrentMonth();
+            if (planParams.vestingIntervalTimeUnit == TimeUnit.DAY) {
+                return _daysBetweenMonths(firstVestingMonth, currentMonth)
+                    .add(
+                        now
+                            .sub(timeHelpers.monthToTimestamp(currentMonth))
+                            .div(_SECONDS_PER_DAY)
+                    )
+                    .div(planParams.vestingInterval);
+            } else if (planParams.vestingIntervalTimeUnit == TimeUnit.MONTH) {
+                return currentMonth
+                    .sub(firstVestingMonth)
+                    .div(planParams.vestingInterval);
+            } else if (planParams.vestingIntervalTimeUnit == TimeUnit.YEAR) {
+                return currentMonth
+                    .sub(firstVestingMonth)
+                    .div(_MONTHS_PER_YEAR)
+                    .div(planParams.vestingInterval);
+            } else {
+                revert("Unknown time unit");
+            }
         }
-        uint256 dateTime = _getTimePointInCorrectPeriod(date, planParams.vestingStepTimeUnit);
-        uint256 lockupTime = _getTimePointInCorrectPeriod(
-            timeHelpers.addMonths(beneficiaryPlan.startMonth, planParams.vestingCliff),
-            planParams.vestingStepTimeUnit
-        );
-        return dateTime.sub(lockupTime).div(planParams.vestingStep);
     }
 
     /**
      * @dev Returns the number of total vesting events.
      */
     function _getNumberOfAllVestingEvents(address wallet) internal view returns (uint) {
-        ITimeHelpers timeHelpers = ITimeHelpers(contractManager.getContract("TimeHelpers"));
         Beneficiary memory beneficiaryPlan = _beneficiaries[wallet];
         Plan memory planParams = _plans[beneficiaryPlan.planId - 1];
-        uint256 finishTime = _getTimePointInCorrectPeriod(
-            timeHelpers.addMonths(beneficiaryPlan.startMonth, planParams.totalVestingDuration),
-            planParams.vestingStepTimeUnit
-        );
-        uint256 afterLockupTime = _getTimePointInCorrectPeriod(
-            timeHelpers.addMonths(beneficiaryPlan.startMonth, planParams.vestingCliff),
-            planParams.vestingStepTimeUnit
-        );
-        return finishTime.sub(afterLockupTime).div(planParams.vestingStep);
+        if (planParams.vestingIntervalTimeUnit == TimeUnit.DAY) {
+            return _daysBetweenMonths(
```

Comment:
addPlan(1, 2, 0, 2, true, true) - in second month 30 days
So 1 vesting would after 1 month - lockupAmount would be vested
Then every 2 days would be vesting - 30 / 2 = 15
Total is equal 16
Are you calculate lockupPeriodEnd Vesting as part of all Vesting events?

Reply:
No. lockup end is not considered as vesting event.
I agree that it may be confusing but this is internal function so let's just assume that lockup end is not a vesting event. As for me it make computations simpler.

Reply:
Good!